File: src/server/src/main/java/com/project/application/models/tag/TagViewResource.java
package com.project.application.models.tag;

import com.project.domain.post.Tag;
import io.micronaut.serde.annotation.Serdeable;

import java.util.UUID;

@Serdeable
public record TagViewResource(UUID id, String name, String translatedName) {
    public static TagViewResource from (Tag tag) {
        return new TagViewResource(tag.getId(), tag.getName(), tag.getTranslatedName());
    }
}

File: src/server/src/main/kotlin/com/project/application/models/tag/TagViewResource.kt
package com.project.application.models.tag

import com.project.domain.post.Tag
import io.micronaut.serde.annotation.Serdeable
import java.util.UUID

@Serdeable
data class TagViewResource(val id: UUID?, val name: String?, val translatedName: String?) {
    companion object {
        fun from(tag: Tag): TagViewResource {
            return TagViewResource(tag.id, tag.name, tag.translatedName)
        }
    }
}
